src/main/java/seedu/exercise/storage/serializablebook/SerializableResourceBook.java

```diff
@@ -0,0 +1,47 @@
+package seedu.exercise.storage.serializablebook;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import seedu.exercise.commons.exceptions.IllegalValueException;
+import seedu.exercise.model.Resource;
+import seedu.exercise.model.book.ReadOnlyResourceBook;
+import seedu.exercise.storage.resource.JsonAdaptedResource;
+
+/**
+ * An immutable ResourceBook that is serializable to JSON format.
+ * In particular, this resource book can be extended to create an immutable ResourceBook that holds
+ * any {@code JsonAdaptedResource} of type {@code T}.
+ */
+public class SerializableResourceBook<T extends JsonAdaptedResource> {
+
+    public static final String MESSAGE_DUPLICATE_RESOURCE = "The list has duplicate exercises/regimes/schedules.";
+
+    protected final List<T> jsonResources = new ArrayList<>();
```

Comment:
Why is the access modifier here `protected`?

Reply:
Changed it to private